overlord/hookstate/ctlcmd/get.go

```diff
@@ -208,7 +212,16 @@ func interfaceHookType(hookName string) (ifaceHookType, error) {
 		return prepareSlotHook, nil
 	} else if strings.HasPrefix(hookName, "connect-slot-") {
 		return connectSlotHook, nil
+	} else if strings.HasPrefix(hookName, "disconnect-plug-") {
```

Comment:
(nitpick^2) as a drive-by comment, this function could be written as:
```
func interfaceHookType(hookName string) (ifaceHookType, error) {
	switch {
	case strings.HasPrefix(hookName, "prepare-plug-"):
		return preparePlugHook, nil
	case strings.HasPrefix(hookName, "connect-plug-"):
		return connectPlugHook, nil
	case strings.HasPrefix(hookName, "prepare-slot-"):
		return prepareSlotHook, nil
	case strings.HasPrefix(hookName, "connect-slot-"):
		return connectSlotHook, nil
	case strings.HasPrefix(hookName, "disconnect-plug-"):
		return disconnectPlugHook, nil
	case strings.HasPrefix(hookName, "disconnect-slot-"):
		return disconnectSlotHook, nil
	case strings.HasPrefix(hookName, "unprepare-slot-"):
		return unprepareSlotHook, nil
	case strings.HasPrefix(hookName, "unprepare-plug-"):
		return unpreparePlugHook, nil
	default:
		return unknownHook, fmt.Errorf("unknown hook type")
	}
}
```
Not sure if its worth doing (or actually really nicer than what is there). But I wanted to mention it :)

Reply:
Good idea, refactored it, thanks!